### ISSUE TITLE
Add auto-prefixing of project stylesheets

### ISSUE THREAD

**Issue by vbalcom** [SapientNitroLA/sugarcoat#25]
- [x] Define method of consuming assets
- [x] Prefix assets
- [x] Define method of inserting prefixed assets into head.hbs partial
- [x] Refine sugarcoat's selectors to be specific for every tagname
- [x] Test feature

**Comment by vbalcom** [comment]
We require two additional options for prefixing: `prefixSelector` and the `assets` to be prefixed. If we include the assets in current template.assets array, we will have to do file type detection, thus mixing modified and non-modified assets together. `prefixSelector` option should be included somehow for if the user creates their own pattern library templates, they should not be restricted to using our default prefix (`.sugar-example`). 

With these two options included in our current config, it would be extended as such: 

```js
var config = {
    settings: {
        dest: './documentation', 
        title: 'Pattern Library',
        template: {
            partials: [
                'documentation/template'
            ],
            assets: [
                'sugarcoat', //  default pattern library styles and js  modules
                'library/css', // custom css
                'library/images'
            ],
            prefixSelector: '.sugar-example' // default
        }
    }
}
```

In order to compartmentalize the prefixing feature better, I'm proposing the following configuration:

```js
var config = {
    settings: {
        dest: './documentation', 
        title: 'Pattern Library',
        template: {
            partials: [
                'documentation/template'
            ],
            // non-modified assets
            assets: [
                'sugarcoat', // default pattern library styles and js  modules
                'library/images'
            ]
        },
        prefix: {
            // modified assets
            assets: [
                'library/css' // custom css
            ],
            selector: '.sugar-example' // default
        }
    }
}
```

**Comment by tisAliG** [comment]
Merged #36, closing.
